src/containers/Tenant/Schema/SchemaViewer/SchemaViewer.tsx

```diff
@@ -81,26 +153,54 @@ export const SchemaViewer = ({keyColumnIds = [], columns = [], type}: SchemaView
                 return undefined;
             },
         },
-    ];
+    );
 
-    if (isExternalTable(type)) {
-        // External tables don't have key columns
-        dataTableColumns = dataTableColumns.filter(
-            (column) => column.name !== SchemaViewerColumns.key,
+    if (withFamilies && isRowTable(type)) {
```

Comment:
Lets move columns calculation into separate function in `./helpers`.

Reply:
It seems to me that this is part of the view logic, I want to see it in the component code in order to understand what I am rendering.

Reply:
Allover our codebase we calculate columns in helpers, that's why I suggested this. And as for me, this logic should be separated for easier reading - maybe not move to separate file, but to the separate function definitely.

Reply:
Do you mean the `withFamilies && isRowTable(type)` condition in separate function or all columns logic with `push`es?

Reply:
I mean all columns logic!

Reply:
If it looks like
```tsx
<Table>
  <Column name="a" />
  {withFamilies && <Column name="b" />}
</Table>
```
do we need a helper?

For me, we have the same but in objects, not jsx.

Reply:
I've meant something like this:
```
function getSchemaColumns(type?: EPathType, withFamilies?: boolean) {
...
}
```

Reply:
```ts
function getSchemaColumns(type?: EPathType, withFamilies?: boolean, families: Record<number, TFamilyDescription> = {}, keyColumnIds: number[] = []) {
...
}
```

Looks like the `SchemaViewer` itself.

Reply:
As for me `SchemaViewer` is a presenter, and `getSchemaColumns` incapsulates logic.

Reply:
Done.